### PR TITLE
validation of olm bundles part1

### PR DESCRIPTION
**Pull request by Avni-Sharma**
### Motivation

Refers - https://issues.redhat.com/browse/APPSVC-392

### Changes

[WIP]

### Testing

Pending

### CODE REVIEW

File: test/operator-hub/catalog_source.yaml [hunk 1]
```diff
@@ -6,6 +6,6 @@ metadata:
   namespace: openshift-operator-lifecycle-manager
 spec:
   sourceType: grpc
-  image: REPLACE_IMAGE
+  image: quay.io/redhat-developer/app-binding-operator:4c0a725
```
Comment (baijum) [inline test/operator-hub/catalog_source.yaml]: Why tag is hard-coded here?

Comment (baijum) [reply test/operator-hub/catalog_source.yaml]: I don't see this file is being used in the Makefile. Do we really need the catalog source?

File: test/operator-hub/subscription.yaml [hunk 2]
```diff
@@ -2,10 +2,11 @@
...
   namespace: openshift-operators
 spec:
-  channel: dev
-  name: service-binding
-  source: my-catalog
-  sourceNamespace: openshift-operator-lifecycle-manager
+  channel: alpha
+  installPlanApproval: Automatic
+  name: service-binding-operator
+  source: example-operators
```
Comment (baijum) [inline test/operator-hub/subscription.yaml]: In `./test/operator-hub/operator_source.yaml`  the name is given as `my-catalog`. Can you please confirm if this is correct? Also, there is a `REPLACE_IMAGE` in that file. When is that value getting replaced?

Comment (baijum) [reply test/operator-hub/subscription.yaml]: Please ignore the above comment. That file doesn't exist. But it is referred from Makefile.

File: Makefile [hunk 3]
```diff
@@ -455,5 +455,19 @@ push-bundle-to-quay:
 .PHONY: dev-release
 ## validating the operator by installing new quay releases
 dev-release:
-	echo  "HELLO BUNDLE VERSION -${BUNDLE_VERSION}"
-	sleep 10m
+	$(Q)kubectl delete operatorsources community-operators -n=openshift-marketplace
+	# Listing of the operator
+	$(Q)kubectl apply -f ./test/operator-hub/operator_source.yaml
```
Comment (baijum) [inline Makefile]: This file doesn't exist. Did you forget to commit it?

File: Makefile [hunk 4]
```diff
@@ -455,5 +455,19 @@ push-bundle-to-quay:
 .PHONY: dev-release
 ## validating the operator by installing new quay releases
 dev-release:
-	echo  "HELLO BUNDLE VERSION -${BUNDLE_VERSION}"
-	sleep 10m
+	$(Q)kubectl delete operatorsources community-operators -n=openshift-marketplace
```
Comment (sbose78) [inline Makefile]: You sure the CVO operator which manages this wouldn't bring this resource back? 🤕

File: Makefile [hunk 5]
```diff
@@ -455,5 +455,19 @@ push-bundle-to-quay:
...
+	# Listing of the operator
+	$(Q)kubectl apply -f ./test/operator-hub/operator_source.yaml
+	# Subscribing to the operator
+	$(Q)kubectl apply -f ./test/operator-hub/subscription.yaml
+	$(eval VERSION_NUMBER := $(shell k get csvs  -n=default -o jsonpath='{.items[*].spec.version}'))
```
Comment (sbose78) [inline Makefile]: You might also want to check if installation has succeeded?

Comment (baijum) [inline Makefile]: Replace `k` with `kubectl`

**Comment by sbose78** [comment]
How do I validate this PR?

File: Makefile [hunk 6]
```diff
@@ -455,5 +455,19 @@ push-bundle-to-quay:
...
+	@if [ ${VERSION_NUMBER} -eq ${BUNDLE_VERSION} ]; \
```
Comment (baijum) [inline Makefile]: I think you need to use `==` instead of `-eq`.

**Comment by baijum** [comment]
> How do I validate this PR?

The script is designed to run on master branch. It's counting the commits for validation. So, you can squash the commits in this PR and run the Makefile target against any cluster.

**Comment by Avni-Sharma** [comment]
> How do I validate this PR?

https://prow.svc.ci.openshift.org/job-history/origin-ci-test/logs/periodic-ci-redhat-developer-service-binding-operator-master-dev-release

**Pull request merged**